### PR TITLE
Switch to docstring_parser to support additional docstring styles

### LINKED ISSUE #16: Support other docstring types
Only Google style docstrings are currently supported:
https://github.com/jdidion/autoclick/blob/05dd571e212672d6d793b2b74428cf40b505c494/autoclick/core.py#L245-L246

### PR DESCRIPTION
**Pull request by notmatthancock**
Closes #16 via https://github.com/rr-/docstring_parser

### CODE REVIEW

**Comment by notmatthancock** [comment]
Manual test script:
```python
import autoclick


def google_func(a: str, b: int = 42):
    """This is the short description

    with some longer info

    Args:
        a: This is the description of `a`.
        b: This is the description of `b`.
    """


def numpy_func(a: str, b: int = 42):
    """This is the short description

    with some longer info

    Parameters
    ----------
    a: str
        This is the description of `a`.
    b: int
        This is the description of `b`.
    """


def rest_func(a: str, b: int = 42):
    """This is the short description

    with some longer info

    :param a: This is the description of `a`.
    :param b: This is the description of `b`.
    """

command = autoclick.command(
    show_defaults=True,
    positionals_as_options=True,
)
command(numpy_func)()
```

File: pyproject.toml [hunk 1]
```diff
@@ -13,11 +13,11 @@ include = ["CHANGES.md"]
 [tool.poetry.dependencies]
 python = "^3.6"
 click = "7.0+"
-docparse = "*"
+docstring_parser = "^0.8.1"
```
Comment (notmatthancock) [inline pyproject.toml:16]: This actually doesn't work. But the version exists and can be pip installed https://pypi.org/project/docstring-parser/0.8.1/
I haven't used poetry before, so I'm not quite sure why it wouldn't pick it up. Ideas?

Comment (jdidion) [reply pyproject.toml:16]: It works for me:
image

Maybe try doing it within a clean virtualenv?

Comment (notmatthancock) [reply pyproject.toml:16]: Turned out the lock file was in a broken state or something. `poetry install` matches your screenshot after deleting the lock and reinstalling.

**Pull request merged**